hosted/alibaba/p0/p0_suite_test.go

```diff
@@ -76,30 +80,35 @@ func p0NodesChecks(cluster *management.Cluster, client *rancher.Client, clusterN
 		initialNodeCount = *cluster.AliConfig.NodePools[0].DesiredSize
 	} else {
 		Fail("initialNodeCount could not be determined from cluster's spec")
+		if cluster.AliStatus != nil && cluster.AliStatus.UpstreamSpec != nil && len(cluster.AliStatus.UpstreamSpec.NodePools) > 0 && cluster.AliStatus.UpstreamSpec.NodePools[0].DesiredSize != nil {
```

Comment:
Will this and remaining code get executed after the `Fail()` function call on L#82?
I think `Fail` will call panic and the entire test would fail.

Reply:
Yeah right, updated the code.